packages/math/src/shapes/Rectangle.js

```diff
@@ -235,14 +235,22 @@ export default class Rectangle
         }
     }
 
-    round(value)
+    /**
+     * Enlarges rectangle that way its corners lie on grid
+     *
+     * @param {number} [resolution=1] resolution
+     * @param {number} [eps=0.001] precision
+     */
+    ceil(resolution = 1, eps = 0.001)
     {
-        value = value || 1;
+        const x2 = Math.ceil((this.x + this.width - eps) * resolution) / resolution;
```

Comment:
Based on our findings from #4745, I think you should take out this code for ceil. Unnecessary

Reply:
in this particular place its actually needed. This rectangle can be anything and we dont know order of operations for getBounds() , it could be something really devastating.